internal/logger/default_logger.go

```diff
@@ -0,0 +1,38 @@
+// Copyright 2024 Canonical.
+package logger
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/juju/zaputil/zapctx"
+	"github.com/mattn/go-colorable"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zapcore"
+)
+
+// SetupLogger sets up the default logger.
+// The local logger is a colorized plain text logger.
+// The production logger is a JSON structured logger.
+func SetupLogger(ctx context.Context, logLevel string, logLocal bool) {
+	pLogLevel, err := zapcore.ParseLevel(logLevel)
+	if err != nil {
+		fmt.Printf("ERROR: log level '%s' cannot be parsed, defaulting to info\n", logLevel)
```

Comment:
**Hint:** You can use `%q` to automatically quote the value:

```go
fmt.Printf("... %q ...", logLevel)
```

Reply:
i'll miss your hint babak <3